### PR TITLE
Add search feature to secrets page

### PR DESCRIPTION
**Pull request by savindi7**
secret search

### Related Issues
- https://github.com/wso2/product-is/issues/13319

### Checklist
- [ ] e2e cypress tests locally verified.
- [ ] Manual test round performed and verified.
- [ ] UX/UI review done on final implementation.
- [ ] Documentation provided. (Add links if there's any)
- [ ] Unit tests provided. (Add links if there's any)
- [ ] Integration tests provided. (Add links if there's any)

### Related PRs
- None

### Security checks
- [ ] Followed secure coding standards in http://wso2.com/technical-reports/wso2-secure-engineering-guidelines?
- [ ] Ran FindSecurityBugs plugin and verified report?
- [ ] Confirmed that this PR doesn't commit any keys, passwords, tokens, usernames, or other secrets?

### CODE REVIEW

File: apps/console/src/features/secrets/components/secrets-list.tsx [hunk 1]
```diff
@@ -318,16 +448,57 @@ const SecretsList: FC<SecretsListProps> = (props: SecretsListProps): ReactElemen
...
+                                placeholder={ t("console:develop.features.secrets.advancedSearch.placeholder") }
+                                defaultSearchAttribute="name"
+                                defaultSearchOperator="co"
+                                data-testid={ `${ testId }-list-advanced-search` }
+                            />
+                        ) }
+                        currentListSize={ secretList.length }
+                        onPageChange={ () => 
+                            void 0
+                        }
```
Comment (yasinmiran) [inline apps/console/src/features/secrets/components/secrets-list.tsx:483]: Let's format this properly.

```suggestion
                        onPageChange={ () => void 0 }
```

File: apps/console/src/features/secrets/components/secrets-list.tsx [hunk 2]
```diff
@@ -266,6 +310,92 @@ const SecretsList: FC<SecretsListProps> = (props: SecretsListProps): ReactElemen
...
+     * Handles the `onFilter` callback action from the
+     * secrets search component.
+     *
+     * @param {string} query - Search query.
+     */
+    const handleSecretsFilter = (query: string) => {
+
+        setSearchQuery(query);
+
+        if(!query){
```
Comment (yasinmiran) [inline apps/console/src/features/secrets/components/secrets-list.tsx:353]:
```suggestion
        if (!query) {
```

File: apps/console/src/features/secrets/components/secrets-list.tsx [hunk 3]
```diff
@@ -266,6 +310,92 @@ const SecretsList: FC<SecretsListProps> = (props: SecretsListProps): ReactElemen
...
+            setFilteredSecrets(secretList.reverse());
+        }
+        
+        const records =  query?.split(" ");
+
+        if(!records){
```
Comment (yasinmiran) [inline apps/console/src/features/secrets/components/secrets-list.tsx:359]:
```suggestion
        if (!records) {
```

File: apps/console/src/features/secrets/components/secrets-list.tsx [hunk 4]
```diff
@@ -266,6 +310,92 @@ const SecretsList: FC<SecretsListProps> = (props: SecretsListProps): ReactElemen
...
+    };
+
+    /**
+     * Sets the list sorting strategy.
+     *
+     * @param {React.SyntheticEvent<HTMLElement>} event - The event.
+     * @param {DropdownProps} data - Dropdown data.
+     */
+    const handleListSortingStrategyOnChange = (event: SyntheticEvent<HTMLElement>,
+        data: DropdownProps): void => {
```
Comment (yasinmiran) [inline apps/console/src/features/secrets/components/secrets-list.tsx:393]:
```suggestion
    const handleListSortingStrategyOnChange = (
        event: SyntheticEvent<HTMLElement>,
        data: DropdownProps
    ): void => {
```

File: apps/console/src/features/secrets/pages/secrets.tsx [hunk 5]
```diff
@@ -142,14 +145,22 @@ const SecretsPage: FC<SecretsPageProps> = (props: SecretsPageProps): ReactElemen
                         { t("console:develop.features.secrets.page.primaryActionButtonText") }
                     </PrimaryButton>
                 </Show>
-            ) }>
+            )
+            }
```
Comment (yasinmiran) [inline apps/console/src/features/secrets/pages/secrets.tsx:149]:
```suggestion
            ) }
```

**Review by yasinmiran — APPROVED** [review]

File: apps/console/src/features/secrets/components/secrets-list.tsx (hunk 1 above)
Comment (savindi7) [reply apps/console/src/features/secrets/components/secrets-list.tsx]: Fixed with https://github.com/wso2/identity-apps/pull/2946/commits/fd8b8b5fa0fceede98d85ab312f115342b9535bf

File: apps/console/src/features/secrets/components/secrets-list.tsx (hunk 2 above)
Comment (savindi7) [reply apps/console/src/features/secrets/components/secrets-list.tsx]: Fixed with https://github.com/wso2/identity-apps/pull/2946/commits/fd8b8b5fa0fceede98d85ab312f115342b9535bf

File: apps/console/src/features/secrets/components/secrets-list.tsx (hunk 3 above)
Comment (savindi7) [reply apps/console/src/features/secrets/components/secrets-list.tsx]: Fixed with https://github.com/wso2/identity-apps/pull/2946/commits/fd8b8b5fa0fceede98d85ab312f115342b9535bf

File: apps/console/src/features/secrets/components/secrets-list.tsx (hunk 4 above)
Comment (savindi7) [reply apps/console/src/features/secrets/components/secrets-list.tsx]: Fixed with https://github.com/wso2/identity-apps/pull/2946/commits/fd8b8b5fa0fceede98d85ab312f115342b9535bf

File: apps/console/src/features/secrets/pages/secrets.tsx (hunk 5 above)
Comment (savindi7) [reply apps/console/src/features/secrets/pages/secrets.tsx]: Fixed with https://github.com/wso2/identity-apps/pull/2946/commits/fd8b8b5fa0fceede98d85ab312f115342b9535bf

File: apps/console/src/features/secrets/components/secrets-list.tsx [hunk 6]
```diff
@@ -266,6 +310,94 @@ const SecretsList: FC<SecretsListProps> = (props: SecretsListProps): ReactElemen
...
+                    data-testid={ `${ testId }-empty-search-placeholder` }
+                />
+            );
+        }
+
+        return null;
+    };
+
+    /**
+     * Handles the `onFilter` callback action from the
```
Comment (Wathsara) [inline apps/console/src/features/secrets/components/secrets-list.tsx:344]: Do we need a line break for this comment?

Comment (savindi7) [reply apps/console/src/features/secrets/components/secrets-list.tsx:344]: Fixed with https://github.com/wso2/identity-apps/pull/3122/commits/f91e9dcceaefc721392d013e788c15d70e0c4612

**Pull request merged**